web/packages/teleterm/src/ui/commandLauncher.ts

```diff
@@ -55,23 +57,30 @@ const commands = {
         host = parts[0];
       }
 
-      // TODO(ravicious): Handle finding host by more than just a name.
-      // Basically we have to replicate tsh ssh behavior here.
-      const servers = ctx.clustersService.searchServers(localClusterUri, {
-        search: host,
-        searchableProps: ['hostname'],
-      });
       let server: tsh.Server | undefined;
+      let serverUri: ServerUri, serverHostname: string;
 
-      if (servers.length === 1) {
-        server = servers[0];
-      } else if (servers.length > 1) {
+      try {
+        // TODO(ravicious): Handle finding a server by more than just a name.
+        // Basically we have to replicate tsh ssh behavior here.
+        server = await retryWithRelogin(ctx, localClusterUri, () =>
+          ctx.resourcesService.getServerByHostname(localClusterUri, host)
+        );
+      } catch (error) {
         // TODO(ravicious): Handle ambiguous host name. See `onSSH` in `tool/tsh/tsh.go`.
-        console.error('Ambiguous host');
+        if (error instanceof AmbiguousHostnameError) {
+          // Log the ambiguity of the hostname but continue anyway. This will pass the ambiguous
+          // hostname to tsh ssh and show an appropriate error in the new tab.
+          console.error(error.message);
+        } else {
+          ctx.notificationsService.notifyError({
+            title: `Could not establish a connection to ${host}`,
+            description: error['message'],
+          });
+          return;
```

Comment:
Do I understand this correctly? If its an ambiguoushostnameerror we want to let the code continue firing after the console log, but other errors stop the rest of the execution? What is the reasoning behind "allowing" this type of error to be shown in a new tab and not the rest?

Reply:
The original idea when I first implemented that was that we allow `AmbiguousHostnameError` to pass through because we're essentially shifting the responsibility from Connect to tsh. We know that if we attempt to execute `tsh ssh` with the given login and hostname, tsh itself will display an appropriate error so we let it do that as at the time we didn't have a good mode of showing this error in any other way in Connect – our only other option was showing an error notification which we deemed subpar to a failed `tsh ssh` execution in a PTY.

So in the end, we do that because we didn't (have time to) think of a better way to approach this particular failure.

Other errors are not passed through because `resourcesService.getServerByHostname` makes a network call so there's a myriad of things that can go wrong and we cannot continue if we get an unexpected error here.

In other words, `AmbiguousHostnameError` being thrown by `ResourcesService` directly maps to a certain error handling scenario of `tsh ssh`. We cannot say that of other errors that might be thrown by `ResourcesService`.